resource-managers/mesos/src/main/scala/org/apache/spark/deploy/mesos/config.scala

```diff
@@ -56,6 +56,26 @@ package object config {
         .createOptional
   }
 
+  private[spark] val CREDENTIAL_PRINCIPAL =
+    ConfigBuilder("spark.mesos.principal")
```

Comment:
Do you think we had better have `.doc` like the previous conf SECRET_FILENAMES at line 50 ~ 56?